programmers/예산.js

```diff
@@ -0,0 +1,11 @@
+function solution(d, budget) {
+    const dSort = d.sort((a, b) => a - b);
+    let answer = 0;
+    for (let i = 0; i < d.length; i++) {
+        if (dSort[i] <= budget)  { 
+            answer++
+            budget -= dSort[i];
+        }
+    }
+    return answer;
+}
```

Comment:
dSort라는 변수를 새로 초기화하기 보다 그냥 d.sort() 를 하면 자동으로 d가 정렬된 상태로 변화 함

```suggestion
function solution(d, budget) {
    let answer = 0;
    
    d.sort((a, b) => a - b);
    
    for (let i = 0; i < d.length; i++) {
        if (d[i] <= budget)  { 
            answer++
            budget -= d[i];
        }
    }
    return answer;
}
```

Reply:
forEach 메서드를 이용한 풀이도 가능함 (참고)
```js
function solution(d, budget) {
    let answer = 0;
    let sum = 0;

    d.sort((a, b) => (a - b));
    
    d.forEach(el => {
        sum += el;
        if (sum <= budget) answer++;
    });
    
    return answer;
}
```